src/GoTrueClient.ts

```diff
@@ -123,20 +124,22 @@ export default class GoTrueClient {
       headers: settings.headers,
       fetch: settings.fetch,
     })
-    this._recoverAndRefresh()
-    this._handleVisibilityChange()
+
     this.url = settings.url
     this.headers = settings.headers
     this.fetch = resolveFetch(settings.fetch)
 
     if (settings.detectSessionInUrl && isBrowser() && !!getParameterByName('access_token')) {
       // Handle the OAuth redirect
-      this._getSessionFromUrl().then(({ error }) => {
+      this.gettingSessionFromUrlPromise = this._getSessionFromUrl().then(({ error }) => {
         if (error) {
           throw new AuthUnknownError('Error getting session from URL.', error)
```

Comment:
Unfortunately this error can't be caught and it might break an app: see this [codesandbox](https://codesandbox.io/s/getsessionfromurl-error-txi7p6?file=/src/index.ts)
The only way to return async errors from a constructor is via a promise.